apps/prairielearn/src/tests/questionSharing.test.ts

```diff
@@ -458,5 +484,91 @@ describe('Question Sharing', function () {
       );
       assert(publiclySharedQuestionRes.ok);
     });
+
+    step('Fail to sync if shared question is renamed', async () => {
+      const questionPath = path.join(sharingCourse.path, 'questions', SHARING_QUESTION_QID);
+      const questionTempPath = questionPath + '_temp';
+      await fs.rename(questionPath, questionTempPath);
+      await syncFromDisk.syncOrCreateDiskToSql(sharingCourse.path, logger);
+      const question_id = await sqldb.queryOptionalRow(
+        sql.get_question_id,
+        {
+          course_id: sharingCourse.id,
+          qid: SHARING_QUESTION_QID,
+        },
+        IdSchema,
+      );
+      assert(
+        question_id !== null,
+        'Sync of consuming course should not allow renaming a shared question.',
+      );
+      await fs.rename(questionTempPath, questionPath);
+    });
+
+    step('Sync through the sync page to extablish commit hash.', async () => {
```

Comment:
Consider just calling `updateCourseCommitHash` instead, if all you want is to get the commit hash into the DB.

Reply:
A secondary reason for this is making sure that I can Successfully sync the course through the sync page, so that in the next test when it fails, I know that it failed for the correct reason. I think I will leave this but change the message to 'Ensure sync through sync page succeeds before editing shared question' or something like that.